### PR TITLE
[fastText] Remove the pointer and improve the build process

### PR DESCRIPTION
**Pull request by stu1130**
## Channel for questions ##
Before or while filing an issue please feel free to join our [Slack channel](https://join.slack.com/t/deepjavalibrary/shared_invite/zt-ar91gjkz-qbXhr1l~LFGEIEeGBibT7w) to get in touch with development team, ask questions, find out what's cooking and more!

## Description ##
(Brief description of what this PR is about)

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- [ ] Code is well-documented: 
    - For user-facing API changes, Java doc has been updated. 
    - For new examples, README.md is added to explain the what the example does.
- [ ] To the my best knowledge, [examples](https://github.com/awslabs/djl/tree/master/examples) and [jupyter notebooks](https://github.com/awslabs/djl/tree/master/jupyter) are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, Java doc)
- [ ] Feature2, tests, (and when applicable, Java doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made?
- Interesting edge cases to note here

### CODE REVIEW

File: extensions/fasttext/src/main/java/ai/djl/fasttext/jni/FtWrapper.java [hunk 1]
```diff
@@ -87,9 +87,9 @@ public void runCmd(String[] args) {
     /** {@inheritDoc} */
     @Override
     public void close() {
-        Pointer pointer = handle.getAndSet(null);
-        if (pointer != null) {
-            FastTextLibrary.LIB.freeFastText(pointer);
+        if (handle != null) {
```
Comment (frankfliu) [inline extensions/fasttext/src/main/java/ai/djl/fasttext/jni/FtWrapper.java:90]:
```suggestion
        Long pointer = handle.getAndSet(null);
        if (pointer != null) {
```

File: extensions/fasttext/src/main/java/ai/djl/fasttext/jni/FtWrapper.java [hunk 2]
```diff
@@ -87,9 +87,9 @@ public void runCmd(String[] args) {
...
+            FastTextLibrary.LIB.freeFastText(handle.get());
```
Comment (frankfliu) [inline extensions/fasttext/src/main/java/ai/djl/fasttext/jni/FtWrapper.java:91]: check handle == null is not atomic, and can cause double free

File: extensions/fasttext/src/main/java/ai/djl/fasttext/jni/NativeResource.java [hunk 3]
```diff
@@ -24,13 +24,11 @@
 
     private static final Logger logger = LoggerFactory.getLogger(NativeResource.class);
 
-    protected final AtomicReference<Pointer> handle;
-    private String uid;
+    protected AtomicLong handle;
```
Comment (frankfliu) [inline extensions/fasttext/src/main/java/ai/djl/fasttext/jni/NativeResource.java:27]: This has to be AtomicReference&lt;Long>

**Review by frankfliu — CHANGES_REQUESTED** [review]

**Review by frankfliu — APPROVED** [review]

**Review by lanking520 — APPROVED** [review]
